src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -2008,15 +2008,28 @@
  * @since 1.5
  */
 public final class Formatter implements Closeable, Flushable {
+
+    // Caching DecimalFormatSymbols
+    static DecimalFormatSymbols DFS = null;
+    static DecimalFormatSymbols getDecimalFormatSymbols(Locale locale) {
+        DecimalFormatSymbols dfs = DFS;
+        if (dfs != null && dfs.getLocale() == locale) {
```

Comment:
`Locale` should be compared using `equals()`.

Reply:
I know it looks wrong and I debated with myself about it, but 1) Locale equals is complex 2) many Locales are global constants 3) there is a 1-1 correspondence of DecimalFormatSymbols to locale. AFAIK even If two locales describe the same configuration there will two distinct DecimalFormatSymbols. Is this not the case? I can add a comment to indicate that this is was deliberate decision.

Reply:
I am afraid people are still using constructors for creating a locale, instead of the factory method that was added later. Since `new Locale("en") == new Locale("en")` returns `false`, I'd still expect `equals()` to compare locales. As to the constants, the number of them is relatively small, IMO.

Reply:
As a separate/future issue, perhaps the constructors should be deprecated to nudge people to using the static `getInstance` methods.

Reply:
Would it be just as effective and improve performance more broadly to cache in DecimalFormatSymbols.getInstance()?

Declarations should be private unless there is a package need.
In this case, the only access to should be via the method.

Reply:
Will investigate

Reply:
Performance is on par when cached in DecimalFormatSymbols. Will switch to that direction.